Buildings/Fluid/HeatExchangers/BaseClasses/WetCoilDryWetRegime.mo

```diff
@@ -0,0 +1,412 @@
+        rotation=0,
+        origin={150,-60})));
+  Modelica.SIunits.HeatFlowRate QLat_flow  "Latent heat transfer rate";
+
+  Modelica.Blocks.Interfaces.RealOutput mCon_flow(
+    quantity="MassFlowRate",
+    final unit="kg/s")
+    "Mass flow of the condensate, negative for dehumidification"
+    annotation (
+      Placement(transformation(extent={{-10,-10},{10,10}},
+        rotation=0,
+        origin={150,-100})));
+  Buildings.Fluid.HeatExchangers.BaseClasses.WetCoilDryRegime fullydry(
+    UAWat=UAWat,
+    mWat_flow=mWat_flow,
+    cpWat=cpWat,
+    TWatIn=TWatIn,
+    UAAir=UAAir,
+    mAir_flow=mAir_flow,
+    mWatNonZer_flow=mWatNonZer_flow,
+    mAirNonZer_flow=mAirNonZer_flow,
+    cpAir=cpAir,
+    TAirIn=TAirIn,
+    final cfg=cfg,
+    mAir_flow_nominal=mAir_flow_nominal,
+    mWat_flow_nominal=mWat_flow_nominal) "Fully-dry coil model";
+
+  Buildings.Fluid.HeatExchangers.BaseClasses.WetCoilWetRegime fullywet(
+    UAWat=UAWat,
+    mWat_flow=mWat_flow,
+    cpWat=cpWat,
+    TWatIn=TWatIn,
+    UAAir=UAAir,
+    mAir_flow=mAir_flow,
+    mWatNonZer_flow=mWatNonZer_flow,
+    mAirNonZer_flow=mAirNonZer_flow,
+    cpAir=cpAir,
+    TAirIn=TAirIn,
+    final cfg=cfg,
+    mAir_flow_nominal=mAir_flow_nominal,
+    mWat_flow_nominal=mWat_flow_nominal,
+    pAir=pAir,
+    X_wAirIn=X_wAirIn) "Fully-wet coil model";
+
+protected
+  Modelica.SIunits.MassFlowRate mAirNonZer_flow(min=Modelica.Constants.eps)=
+    Buildings.Utilities.Math.Functions.smoothMax(
+      x1=mAir_flow,
+      x2=1E-3       *mAir_flow_nominal,
+      deltaX=0.25E-3*mAir_flow_nominal)
+    "Mass flow rate of air";
+  Modelica.SIunits.MassFlowRate mWatNonZer_flow(min=Modelica.Constants.eps)=
+    Buildings.Utilities.Math.Functions.smoothMax(
+      x1=mWat_flow,
+      x2=1E-3       *mWat_flow_nominal,
+      deltaX=0.25E-3*mWat_flow_nominal)
+    "Mass flow rate of water";
+
+  Modelica.SIunits.Temperature TAirInDewPoi
+    "Dew point temperature of incoming air";
+
+  Buildings.Utilities.Psychrometrics.pW_X pWIn(X_w=X_wAirIn,p_in=pAir);
+  Buildings.Utilities.Psychrometrics.TDewPoi_pW TDewIn(p_w=pWIn.p_w);
+
+  //-- parameters for fuzzy logics
+  Real mu_FW(final unit="1", min=0, max=1), mu_FD(unit="1",min=0, max=1)
+    "Membership functions for Fully-Wet and Fully-Dry conditions";
+  Real w_FW(final unit="1", min=0, max=1),  w_FD(unit="1",min=0, max=1)
+    "Normailized weight functions for Fully-Wet and Fully-Dry conditions";
+  Real dryFra(final unit="1", min=0, max=1)
+    "Dry fraction, e.g., 0.3 means condensation occurs at 30% HX length from air inlet";
+
+equation
+
+  TAirInDewPoi=TDewIn.T;
+
+  mu_FW= Buildings.Utilities.Math.Functions.spliceFunction(
+  pos=0,neg=1,x=fullywet.TSurAirIn-TAirInDewPoi,deltax=max(abs(fullydry.TSurAirOut- fullywet.TSurAirIn),1e-3));
+  mu_FD= Buildings.Utilities.Math.Functions.spliceFunction(
+  pos=1,neg=0,x=fullydry.TSurAirOut-TAirInDewPoi,deltax=max(abs(fullydry.TSurAirOut- fullywet.TSurAirIn),1e-3));
```

Comment:
Here you need to use `smoothMax` in the argument `deltax` because `max` is not differentiable, and hence `mu_*` and the other calculations won't be differentiable either.

Reply:
Thanks for pointing this out. It should be soothMax.